### PR TITLE
Add versioneer

### PR DESCRIPTION
**Pull request by msmttchr**
Draft code to evaluate impact of using versioneer.
This can be useful for daily build and to identify intermediate versions between official releases.

Also it could help simplifying release process, but I am not sure.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -8,4 +9,5 @@ recursive-include tests *
 recursive-include docs *
 prune docs/_build
 recursive-exclude * __pycache__
-recursive-exclude * *.pyc
\ No newline at end of file
+recursive-exclude * *.pycinclude
```
Comment (bilderbuchi) [inline MANIFEST.in:12]: Was the L12 edit intentional?

Comment (msmttchr) [reply MANIFEST.in:12]: I think this is a mistake. Thanks for spotting it.

**Comment by bilderbuchi** [comment]
We should compare this approach with using setuptools_scm, which seems to require [much fewer/less invasive changes](https://github.com/Galvant/InstrumentKit/pull/322/files) to achieve something similar?

**Comment by msmttchr** [comment]
> We should compare this approach with using setuptools_scm, which seems to require [much fewer/less invasive changes](https://github.com/Galvant/InstrumentKit/pull/322/files) to achieve something similar?

Possibly, I have no experience with this. I know that versioneer is working and my colleague is doing a daily build with Jenkins, but I don't know what is behind it.

**Comment by bilderbuchi** [comment]
A quick skim of the [readme](https://pypi.org/project/setuptools-scm/) suggests that setuptools_scm can do ~ the same things that versioneer does, without needing to vendor those files (avoiding this would be great imo). 
I'll try this out locally and report back.

**Comment by bilderbuchi** [comment]
> A quick skim of the [readme](https://pypi.org/project/setuptools-scm/) suggests that setuptools_scm can do ~ the same things that versioneer does, without needing to vendor those files (avoiding this would be great imo).
> I'll try this out locally and report back.

I've started work on this in https://github.com/bilderbuchi/pymeasure/tree/dynamic-version (not yet operational) -- probably this will trigger a long-overdue transition to pyproject.toml at the same time.

**Comment by msmttchr** [comment]
As #613 will address similar needs in a preferred way, I will close this draft PR.

**Pull request closed without merging**